### DISCUSSION TITLE
Persistence with Linux Mint 20.3 Cinnamon ? Someone had it ?

### DISCUSSION THREAD

**Discussion by Menard01** [ventoy/Ventoy#1445] [Q&A] [answered]
Hi 
I only want to know if there was a successfull attempt with persistence with Ventoy, personnaly it failed for me with LM20.3 cinnamon iso file, works but without persistence 
Is it normal I had no menu when shuting down ? except the warning to remove the USB Stick 
And when I set the parameters, thre was 2 files, one was a .bat file and the other a json file but these files were entirely created with the web page interface, I added nothing more in it, is this enough ?

I am with legacy bios, mbr partitions, AMD x86-64, radeon and Ventoy 1.0.66 + plugin

**Comment by steve6375** [comment]
If the ventoy.json file is configured correctly, when you boot to Ventoy and select the ISO file, then Ventoy should prompt you with a second menu which asks you if you want to boot without persistence or using the persistence file that you have created.
Did you see this secondary menu?
Did you create the persistence file and use the correct type  (casper-rw) ?
https://ventoy.net/en/plugin_persistence.html

**Comment by Menard01** [comment]
Yes it is configured correctly according the web interface, all is in green, ad yes I ve got this menu and I selected the file that the plugin created, personnaly I only chose a specific name for it as explained 
What do you call a persistence file ? json ? I used the plugin so it is automatically created with only a few choices and casper-rw was the item pre-selected for my system 
I only followed the web interface letting the default settings except when necessary to change  

Oh I discovered I didn't do anything as a so called "backend image file" for me the process was finished at the end of the web interface page, it is no mention of this in it, if I remember well
So am I supposed to download such a file and where to install it, which path ?  It is not clear
Why do you not answer my question ? Did someone had persistence with 20.3 ?

**Comment by Menard01** [comment]
Oh, sorry It was clear : "The backend image file must be put in the first partition. It can be in the root directory or any subdirectory"

**Comment by Menard01** [comment]
"The file with persistence label has a persistence.conf file inside it" The file content is / union
I don't understand this ... 
Did someone succeed with persistence with Linux Mint 20.3 ?

**Comment by steve6375** [comment, marked as answer]
Read the persistence page.
Did you extract and use the correct file for Linux mint? See the table.
You can download pre-created image files from https://github.com/ventoy/backend/releases

**Reply by Menard01** [reply]
I ve just tried another time with the file called "backend image file" and it failed again 
Conclusion : no persistence with LM 20.3 ?

**Reply by steve6375** [reply]
Maybe you should get my ebook on 'Getting started with Ventoy' if you can't follow the instructions on the Ventoy persistence web page?

**Reply by Menard01** [reply]
Sorry but I am not a developper, I don't want to lose more time with a feature I don't know if it  can work with my iso , it is more simple fore me to use an older iso

**Reply by chatmandu-uk** [reply]
> Read the persistence page. Did you extract and use the correct file for Linux mint? See the table. You can download pre-created image files from https://github.com/ventoy/backend/releases

I have added the 'casper-rw' .dat file and created a .json file. 

This passes the persistence validation, but I don't get the option to start Linux Mint with persistence. 

![2022-06-07 13 37 03](https://user-images.githubusercontent.com/107044828/172382010-538ad4ed-6cb5-4579-aeef-93893dc02b8b.jpg)

```
{
 
  "control": [
        { "VTOY_DEFAULT_MENU_MODE": "1" },
        { "VTOY_FILT_DOT_UNDERSCORE_FILE": "1" },
        { "VTOY_DEFAULT_SEARCH_ROOT": "/ISO/" }
    ],
    "persistence" : [
        {
            "image": "/iso/linuxmint-20.3-cinnamon-64bit.iso",
            "backend": "/persistence/persistence_xfs_2GB_casper-rw-linux-mint-20_3.dat"
        }
    ]
}
```

**Edit** 

The issue was due to the 'iso' paths being case sensitive.

**Comment by steve6375** [comment]
![image](https://user-images.githubusercontent.com/1469350/154836501-a19ca320-1f61-4d24-8872-e6535ea79302.png)
![image](https://user-images.githubusercontent.com/1469350/154836683-f6de69fd-2d00-408b-aa00-a00537f19f66.png)

![image](https://user-images.githubusercontent.com/1469350/154836516-0a562657-f730-40e4-b010-58c52fba0684.png)
![image](https://user-images.githubusercontent.com/1469350/154836613-f36a797e-4f75-4686-83b5-f443ae421675.png)


Yes, I can confirm that persistence works (VBox 5, UEFI64).

**Comment by Menard01** [comment]
OK, thanks a lot, but knowing I am not with VBox and UEFI 
I can confirm I have this persistence_ext4_256MB_casper-rw.dat in a subfolder I called "data" but it doesn't work

**Comment by steve6375** [comment]
Unless you post screenshots like the ones I posted, we cannot help.
We need to know exactly the files, their location. your PC model, your USB model, whether booting via BIOS, UEFI32, UEFI64 and your ventoy.json contents.
It clearly works for me and many other people, so you have done something wrong or your USB drive is faulty.

**Comment by Menard01** [comment]
I begin to understand what may have happened because in my case I have 2 dat.files 
I followed all the instructions but perhaps I made a mix because the plugin web interface created a dat file that is empty and the json, both in Ventoy/ventoy ok, but then I followed the instructions to download this other dat file I've pasted in /ventoy/data

I mean I didn't post screenshots because as I said, all was right in the process. What has been useful was to read the json ... to see what happens

**Comment by steve6375** [comment]
The VentoyPlugson web app does NOT create a .dat file !!!
You must extract the .dat file from the download or make it yourself.
The VentoyPlugson web app only allows you to set the path of the .dat file.

**Comment by Menard01** [comment]
But I allways had an empty .dat file with a name I personnaly chose, I don't remember exactly how ... what I am sure is I cannot have invented this sort of file creation ... So now after editing the .json to have the good path to the downloaded .dat file it s ok, it works !!  Youpi !! :)
